src/reachy2_sdk/parts/goto_based_part.py

```diff
@@ -80,3 +83,14 @@ def _is_goto_finished(self, id: GoToId) -> bool:
             or state.goal_status == GoalStatus.STATUS_SUCCEEDED
         )
         return result
+
+    def _wait_goto(self, id: GoToId, timeout: float = 10) -> None:
+        """Wait for a goto to finish. timeout is in seconds."""
+        self._logger_goto.info(f"Waiting for movement with {id}.")
+        t1 = time.time()
+        while not self._is_goto_finished(id):
+            time.sleep(0.1)
+            if time.time() - t1 > timeout:
```

Comment:
I think timeout for the gotos should be checked once the movement has begun only.

If we do:
```
pose2 = get_pose_matrix([0.4, -0.2, -0.1], [0, -90, 0])
reachy.goto_posture("default")
reachy.r_arm.goto_from_matrix(pose2, wait=True)
```
Then the timeout is triggered, whereas the goto was just queued

Reply:
The timeout is now the sum of the duration of all pending movements on the part